fe2-android/app/src/androidTest/java/com/github/dedis/popstellar/ui/detail/event/consensus/ElectionStartFragmentTest.java

```diff
@@ -0,0 +1,278 @@
+  private static final String instanceId =
+      Consensus.generateConsensusId(key.getType(), key.getId(), key.getProperty());
+
+  private static final ConsensusElect elect =
+      new ConsensusElect(pastTime, key.getId(), key.getType(), key.getProperty(), "started");
+  private static final ConsensusElectAccept accept3 =
+      new ConsensusElectAccept(instanceId, "m3", true);
+
+  private static final ConsensusLearn learn3 =
+      new ConsensusLearn(instanceId, "m3", Collections.emptyList());
+
+  private static final ArgumentCaptor<Message> captor = ArgumentCaptor.forClass(Message.class);
+
+  private static String publicKey;
+  private static LAORepository laoRepository;
+  private static boolean isAlreadySetup = false;
+
+  LAORemoteDataSource remoteDataSource = Mockito.mock(LAORemoteDataSource.class);
+  LAOLocalDataSource localDataSource = Mockito.mock(LAOLocalDataSource.class);
+
+  @Rule
+  public final FragmentScenarioRule<ElectionStartFragment> fragmentRule =
+      FragmentScenarioRule.launchInContainer(
+          ElectionStartFragment.class,
+          () -> {
+            ElectionStartFragment fragment = ElectionStartFragment.newInstance();
+            if (!isAlreadySetup) {
+              setup();
+              isAlreadySetup = true;
+            }
+            return fragment;
+          });
+
+  private void setup() {
+    Mockito.when(remoteDataSource.incrementAndGetRequestId()).thenReturn(42);
+    Mockito.when(remoteDataSource.observeWebsocket()).thenReturn(Observable.empty());
+    Observable<GenericMessage> upstream = Observable.fromArray((GenericMessage) new Result(42));
+
+    // Mock the remote data source to receive a response
+    Mockito.when(remoteDataSource.observeMessage()).thenReturn(upstream);
+
+    try {
+      LAORepository.destroyInstance();
+      laoRepository =
+          LAORepository.getInstance(
+              remoteDataSource,
+              localDataSource,
+              Injection.provideAndroidKeysetManager(ApplicationProvider.getApplicationContext()),
+              Injection.provideGson(),
+              new ProdSchedulerProvider());
+      publicKey = getPublicKey();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+
+    election.setStart(futureTime);
+    lao.setChannel(laoChannel);
+    lao.setOrganizer(publicKey);
+    lao.setWitnesses(Sets.newSet(node2Key, node3Key));
+
+    laoRepository.getLaoById().put(laoChannel, new LAOState(lao));
+    laoRepository.updateNodes(laoChannel);
+  }
+
+  @After
+  public void clean() {
+    LAORepository.destroyInstance();
+  }
+
+  private String getPublicKey() throws IOException, GeneralSecurityException {
+    KeysetHandle publicKeysetHandle =
+        Injection.provideAndroidKeysetManager(ApplicationProvider.getApplicationContext())
+            .getKeysetHandle()
+            .getPublicKeysetHandle();
+    return Keys.getEncodedKey(publicKeysetHandle);
+  }
+
+  @Test
+  public void test() throws InterruptedException {
```

Comment:
This test name is not really meaningful.
What are you testing exactly ?

Reply:
I changed it to `displayWithUpdatesIsCorrectAndButtonsProduceCorrectMessages`,
I'm testing that everything is correctly display, with some updates. I also test  that the buttons produces the correct messages.